src/replication.c

```diff
@@ -394,7 +397,9 @@ remove_replica_from_list(spec_t *spec, int iofd) {
 			pthread_cond_signal(&replica->r_cond);
 			MTX_UNLOCK(&replica->r_mtx);
 			spec->replica_count--;
-			if(spec->replica_count == 0){
```

Comment:
Add a TODO statement to decrement health/degraded count based on the state of replica when it got disconnected